sdk/blockchain.go

```diff
@@ -16,98 +16,119 @@ import (
 type BlockchainService service
 
 // Get Block Height
-func (b *BlockchainService) GetBlockByHeight(ctx context.Context, height *big.Int) (*BlockInfo, *http.Response, error) {
+func (b *BlockchainService) GetBlockByHeight(ctx context.Context, height *big.Int) (*BlockInfo, error) {
+	if height == nil || height.Int64() == 0 {
+		return nil, ErrNilOrZeroHeight
+	}
+
 	u := fmt.Sprintf(pathBlockByHeight, height.String())
 
-	bDto := &blockInfoDTO{}
-	resp, err := b.client.DoNewRequest(ctx, "GET", u, nil, &bDto)
+	dto := &blockInfoDTO{}
+
+	resp, err := b.client.DoNewRequest(ctx, http.MethodGet, u, nil, &dto)
 	if err != nil {
-		return nil, resp, err
+		return nil, err
 	}
 
-	bInfo, err := bDto.toStruct()
-	if err != nil {
-		return nil, resp, err
+	if err = handleResponseStatusCode(resp, map[int]error{404: ErrResourceNotFound, 409: ErrArgumentNotValid}); err != nil {
+		return nil, err
 	}
 
-	return bInfo, resp, nil
+	return dto.toStruct()
 }
 
 // Get Transactions from a block information
-func (b *BlockchainService) GetBlockTransactions(ctx context.Context, height *big.Int) ([]Transaction, *http.Response, error) { // TODO Add params
-	u := fmt.Sprintf(pathBlockGetTransaction, height.String())
+func (b *BlockchainService) GetBlockTransactions(ctx context.Context, height *big.Int) ([]Transaction, error) { // TODO Add params
```

Comment:
//todo:

Reply:
Yeah, because of it I have a question, what does this todo mean ?
It was written before and I have no idea which params are missed.

Reply:
Please, will write 'todo' as comment with new line & ':' for IDE undastanding